go/cmd/dolt/commands/merge.go

```diff
@@ -93,146 +100,299 @@ func (cmd MergeCmd) Exec(ctx context.Context, commandStr string, args []string,
 		defer closeFunc()
 	}
 
-	if apr.ContainsAll(cli.SquashParam, cli.NoFFParam) {
-		cli.PrintErrf("error: Flags '--%s' and '--%s' cannot be used together.\n", cli.SquashParam, cli.NoFFParam)
+	if _, ok := queryist.(*engine.SqlEngine); !ok {
+		cli.PrintErrln("`dolt merge` is only supported in local mode.")
```

Comment:
Add to the message that they need to shut down their `dolt sql-server` instance and try again.